tools/np-build-driver-stride.lua

```diff
@@ -0,0 +1,113 @@
+ffi = require 'ffi'
+
+local SLASH = "/"
+if ffi.os == "Windows" then SLASH = "\\" end
+
+-- Code by David Kastrup
+require "lfs"
+
+function dirtree(dir)
+  assert(dir and dir ~= "", "directory parameter is missing or empty")
+  if string.sub(dir, -1) == SLASH then
+    dir=string.sub(dir, 1, -2)
+  end
+
+  local function yieldtree(dir)
+    for entry in lfs.dir(dir) do
+      if entry ~= "." and entry ~= ".." then
+        entry=dir..SLASH..entry
+        local attr=lfs.attributes(entry)
+        coroutine.yield(entry,attr)
+        if attr.mode == "directory" then
+          yieldtree(entry)
+        end
+      end
+    end
+  end
+
+  return coroutine.wrap(function() yieldtree(dir) end)
+end
+
+function string.ends(String,End)
+   return End=='' or string.sub(String,-string.len(End))==End
+end
+
+function string.starts(String,Start)
+   return string.sub(String,1,string.len(Start))==Start
+end
+
+local exclude_dirs = {}
+local exclude_files = {}
+local directory = ""
+
+-- We may need to support both build approaches for now (iOS / android)
+common_flags = "-ffast-math -fno-threadsafe-statics -nostdlibinc -nobuiltininc -nostdinc++ -Wno-macro-redefined -DFLOAT_APPROX -I.."..SLASH.."NativePath -I.."..SLASH.."NativePath"..SLASH.."standard"
+zig_common_flags = "-Wno-ignored-attributes -Wno-delete-non-virtual-dtor -Wno-macro-redefined -fno-rtti -fno-exceptions"
+
+objs = {}
+cfiles = {}
+cppfiles = {}
+hfiles = {}
+debug = false
+is_verbose = false
+is_shared = false
+platform = "windows"
+outputName = "lib"
+
+function table.contains(table, element)
+  for _, value in pairs(table) do
+    if string.starts(element, value) then
+      return true
+    end
+  end
+  return false
+end
+
+for i,v in ipairs(arg) do 
+    if v == "debug" then
+        debug = true
+    elseif string.starts(v, "-I") then
+        local includeDir = string.sub(v, 3)
+        common_flags = common_flags.." -I"..includeDir
+        zig_common_flags = zig_common_flags.." -I"..includeDir
+    elseif string.starts(v, "-D") then
+        local includeDir = string.sub(v, 3)
+        common_flags = common_flags.." -D"..includeDir
+        zig_common_flags = zig_common_flags.." -D"..includeDir
+    elseif string.starts(v, "-n") then
+        local n = string.sub(v, 3)
+        outputName = n
+    elseif string.starts(v, "-E") then
+        local n = string.sub(v, 3)
+        table.insert(exclude_dirs, n)
+    elseif string.starts(v, "-e") then
+        local n = string.sub(v, 3)
+        table.insert(exclude_files, n)
+    elseif v == "--verbose" or v == "-v" then
+        is_verbose = true
+    elseif v == "--shared" then
+        is_shared = true
+    elseif string.starts(v, "-p") then
+        platform = string.sub(v, 3)
+    else
+        directory = v
+    end
+end
+
+for filename, attr in dirtree(directory) do
```

Comment:
Can you add a short comment above this loop what it's doing? It's clear, but it's easier to read through the file by just glancing at the comment

Reply:
I could yes, but this is a copy of the original implementation which also lacked the comments and so I also had to figure it out lol. Will add a comment.